src/CMakeLists.txt

```diff
@@ -343,7 +354,13 @@ if(NOT HPX_WITH_STATIC_LINKING)
     endif()
   endif()
   target_include_directories(hpx_init PRIVATE $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>)
-  target_link_libraries(hpx_init ${HPX_TLL_PUBLIC} hpx_preprocessor)
+
+  foreach(_module ${_modules})
+    # add module include directories as PRIVATE to the hpx_init library to
+    # enable its compilation against indirectly included headers
+    get_target_property(_module_includes ${_module} INTERFACE_INCLUDE_DIRECTORIES)
+    target_include_directories(hpx_init PRIVATE ${_module_includes})
```

Comment:
Do we explicitly _not_ want to link against the modules? Same for `hpx_wrap`.

Reply:
While compiling `hpx_init` (and `hpx_wrap`) the module include directories are needed. Applications using those libraries will see the include directories through the dependency on HPX itself.

Reply:
I meant why would `target_link_libraries(hpx_init PRIVATE ${_module})` not be enough?

Reply:
Fair point. That should do the trick as well. The rationale for my change was that `hpx_init` do not need to _link_ against the modules, they only need to see the includes.

Reply:
Ok. Either is good, just wanted to make sure I understand since this is more code.